[DOC] Update MCI HW and Integration Spec (#365)
chipsalliance/caliptra-ss: docs/CaliptraSSIntegrationSpecification.md

    Each wire connects to a register in the MCI register bank through which communication to the MCU may be facilitated. Each of the generic wire signals is 64 bits in size.These signals are considered ASYNC and each of the 64 bits are considered separate adhoc signals. Meaning there is no bus synchronization which means the connections to this port need to be thoroughly thought through to ensure the MCU doesn’t drop any requests. 

    Activity on any bit of the `mci_generic_input_wires` triggers a notification interrupt to the microcontroller indicating a bit toggle.

    The following table describes the allocation of functionality on `mci_generic_input_wires` . All bits not listed in this table must be tied to 0.

    **Table: MCI Generic Input Allocation** 

    | Bits | Name | Description |
    | :---- | :---- | :---- |

>     | 64:0 | RESERVED | No allocated function |

Reviewer: 63:0?

### Error Aggregation Connectivity Requirements

MCI aggregates all fatal and non-fatal errors for Caliptra SS via two ports `agg_error_fatal` and `agg_error_non_fatal`. These errors are: 

1. Sent to MCU via interrupt
2. Sent to SOC via `all_error_fatal` or `all_error_non_fatal` MCI output ports

Errors connected to this infrastructure are required to be level signals. Pulses are not permitted. 

> MCU has the ability to mask these aggregated interrupts to both its own interrupt along with the `all_error_fatal` `all_error_non_fatal` output port. 

Reviewer: Reword? 
```suggestion
MCU has the ability to independently mask these aggregated interrupts to its own interrupt and to the `all_error_fatal` and `all_error_non_fatal` output ports.
```

Access to MCU SRAM Execution Region is controlled by mcu\_sram\_fw\_exec\_region\_lock. Before MCU is brought out of reset for a hitless patch this signal must be asserted giving access to the MCU. Failure to do so will trigger an AXI access error since Caliptra will be the only entity allowed to access the SRAM until this signal is asserted.

### MCU MBOX SRAM Sizing Requirements

MCU MBOX SRAM size is set via `MCU_MBOX0_SIZE_KB` and `MCU_MBOX1_SIZE_KB` parameters

The MCU can be configured to implement up to 2 independent mailboxes with maximum fully addressable SRAM. sizes of up to 2 MB.  If they are configured sizes of 0, they will not be instantiated.

> Size should be deterined by:

Reviewer: ```suggestion
Size should be deterimed by:
```

### MCU JTAG/DMI Access

![](images/MCI-DMI-Interface.png)

MCI provides access JTAG security for MCU. Detailed debug architecture can be found in [MCI debug spec](https://github.com/chipsalliance/caliptra-ss/blob/main/docs/CaliptraSSHardwareSpecification.md#mci-debug). Some notes about MCI debug arcitecture:

1. MCI registers are accessed via the MCU DMI uncore address space.
2. Limited MCU JTAG access is provided to MCI DMI registers by setting `cptra_ss_debug_intent_i`
3. Full access to MCU core and uncore DMI registers is restricted to LCC unlocked mode. 

#### MCU SRAM JTAG Access

MCU SRAM is fully accessable via MCU JTAG. MCU must be halted before accessing MCU SRAM via JTAG otherwise collisions between AXI and JTAG will result in errors and corrupted data. 

### MCU Trace Buffer

![](images/MCI-MCU-Trace-Buffer-Diagram.png)

MCI hosts the MCU trace buffer. The full trace buffer spec is [here](https://github.com/chipsalliance/caliptra-ss/blob/main/docs/CaliptraSSHardwareSpecification.md#mcu-trace-buffer). High level notes about trace buffer:

1. It can only be accessed when debug unlocked
2. It can be accessed via AXI or DMI

> 3. Traces can be disabled via a Risk-V control register

Reviewer: SP: RISC-V

Author: Updated

1. `cptra_ss_mcu_no_rom_config_i` set to 1
2. `cptra_ss_strap_mci_soc_config_axi_user_i` set to a trusted user other than MCU
3. `cptra_ss_mcu_halt_status_i` connected to `cptra_ss_mcu_halt_status_o` ORed with SOC control
4. `cptra_ss_mcu_halt_ack_i` connected to `cptra_ss_mcu_halt_ack_o` ORed with SOC control
5. (optional) `cptra_ss_strap_mcu_reset_vector_i` set to known starting address in MCU SRAM

The expected boot sequence is:

1. MCI brought out of reset
2. MCI boot FSM progresses to `WAIT_FOR_CPTRA_BOOT_GO`
3. Trusted SOC agent does configuration MCU ROM typically executes. See [CSS HW spec](https://github.com/chipsalliance/caliptra-ss/blob/main/docs/CaliptraSSHardwareSpecification.md#subsystem-boot-finite-state-machine-css-bootfsm)
  - When SOC agent sees `notif_cptra_mcu_reset_req_sts` set by Caliptra, SOC will see `cptra_ss_mcu_halt_req_o` asserted by MCI Boot FSM. SOC must assert `cptra_ss_mcu_halt_status_i` and `cptra_ss_mcu_halt_ack_i` back to MCI. When SOC sees `cptra_ss_mcu_halt_req_o` deassert SOC shall give full control of these signals back to MCU. 
  - See [MCU Halt Ack Interface](#mcu-halt-ack-interface) for recommended connections.

> 4. Trusted SOC agent sets `CPTRA_BOOT_GO.go`

Reviewer: Is this step correct? This boot sequence appears to describe a hitless update for MCU, but why is CPTRA_BOOT_GO set after the MCU reset?

Author: Fixed
